### PR TITLE
Fix validation of time series parameters

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/system_data.jl [hunk 1]
```diff
@@ -549,22 +507,16 @@ end
...
-get_time_series_initial_time(data::SystemData) =
-    get_time_series_initial_time(data.components)
-get_time_series_last_initial_time(data::SystemData) =
-    get_time_series_last_initial_time(data.components)
-get_time_series_horizon(data::SystemData) = get_time_series_horizon(data.time_series_params)
+generate_forecast_initial_times(data::SystemData) =
+    generate_forecast_initial_times(data.time_series_params)
+get_forecast_total_period(data::SystemData) =
+    get_forecast_total_period(data.time_series_params)
+get_forecast_count(data::SystemData) = get_forecast_count(data.time_series_params)
```
Comment (daniel-thom) [inline src/system_data.jl:514]: This term is a bit misleading to me.  Would `get_forecast_window_count` be better?

Comment (jd-lara) [reply src/system_data.jl:514]: ok, we could change it

File: src/system_data.jl [hunk 2]
```diff
@@ -549,22 +507,16 @@ end
...
+get_forecast_horizon(data::SystemData) = get_forecast_horizon(data.time_series_params)
+get_forecast_initial_timestamp(data::SystemData) =
```
Comment (jd-lara) [inline src/system_data.jl:516]: This will imply we can close #141 ?

Comment (daniel-thom) [reply src/system_data.jl:517]: yes

**Review by jd-lara — APPROVED** [review]
LGTM

**Pull request merged**